### PR TITLE
Fix non-stream to stream conversion

### PR DESCRIPTION
**Pull request by djgalloway**
The previous method no longer works

### CODE REVIEW

File: tools/convert-to-centos-stream.yml [hunk 1]
```diff
@@ -8,6 +8,39 @@
...
+      file_type: file
+      patterns: 'CentOS-Linux-*.repo'
+    register: pre_stream_repo_files
+    when: ansible_distribution == 'CentOS'
+
+  - name: Remove all mirrorlists
+    replace:
+      path: "{{ item.path }}"
+      regexp: '^mirrorlist=.*'
+#      replace: 'release=$releasever-stream&'
```
Comment (zmc) [inline tools/convert-to-centos-stream.yml:23]: Mistake?

Comment (djgalloway) [reply tools/convert-to-centos-stream.yml:23]: Shoot.  Yeah.

File: roles/cobbler_profile/tasks/import_distro.yml [hunk 2]
```diff
@@ -8,11 +8,8 @@
   debug: var=distro_name
 
 - name: Load extra_distros from secrets
-  # The jinja2 templating API allows you to update one dict with another,
-  # but it does so in-place without returning any dict. The first few
-  # blobs do the updating; the last actually returns the result.
   set_fact:
-    distros: "{% for distro in distros.keys() + extra_distros.keys() %}{{ distros[distro].update(extra_distros[distro]|default({})) }}{% endfor %}{{ distros }}"
+    distros: "{{ distros|combine(extra_distros, recursive=True) }}"
```
Comment (zmc) [inline roles/cobbler_profile/tasks/import_distro.yml:12]: this is so much nicer

File: tools/convert-to-centos-stream.yml [hunk 3]
```diff
@@ -8,6 +8,39 @@
...
+      regexp: '^\s*#*\s*(baseurl=.*)'
+      replace: '\1'
+    with_items: "{{ pre_stream_repo_files.files }}"
+    when: ansible_distribution == 'CentOS'
+
+  - name: Point baseurls to archive server
+    replace:
+      path: "{{ item.path }}"
+      regexp: 'mirror.centos.org/\$contentdir/\$releasever'
+      replace: 'vault.centos.org/8.5.2111'
```
Comment (zmc) [inline tools/convert-to-centos-stream.yml:40]: This feels like something that could go stale without us really noticing; should it maybe be abstracted to a variable?

**Review by zmc — APPROVED** [review]
Looks good! One concern that isn't a blocker

File: tools/convert-to-centos-stream.yml (hunk 3 above)
Comment (djgalloway) [reply tools/convert-to-centos-stream.yml:40]: Eh.  My thinking was

1. vault.centos.org is supposed to be the vault.  Like it doesn't change.
2. There will never be a newer version of CentOS 8.  8.5.2111 is the last one.

Comment (zmc) [reply tools/convert-to-centos-stream.yml:40]: Oh, of course! 🤦

**Pull request merged**